### PR TITLE
- various improvements

### PR DESCRIPTION
**Pull request by aschnell**

### CODE REVIEW

File: storage/Devices/PartitionTableImpl.cc [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace storage
 
 
     const vector<string> EnumTraits<PtType>::names({
-	"unknown", "loop", "msdos", "gpt", "dasd", "mac"
+	"unknown", "loop", "MSDOS", "GPT", "DASD", "mac"
```
Comment (shundhammer) [inline storage/Devices/PartitionTableImpl.cc]: Consistency? why some lowercase, and those now uppercase?

File: storage/Devices/DiskImpl.cc [hunk 2]
```diff
@@ -107,19 +107,39 @@ namespace storage
     PtType
     Disk::Impl::get_default_partition_table_type() const
     {
-	const Arch& arch = get_devicegraph()->get_storage()->get_arch();
+	PtType ret = get_possible_partition_table_types().front();
```
Comment (shundhammer) [inline storage/Devices/DiskImpl.cc]: the default type is guaranteed to be the first?

File: storage/Devices/DiskImpl.cc [hunk 3]
```diff
@@ -107,19 +107,39 @@ namespace storage
...
+    std::vector<PtType>
+    Disk::Impl::get_possible_partition_table_types() const
+    {
+	// TODO other archs
+
+	const Arch& arch = get_devicegraph()->get_storage()->get_arch();
 
 	unsigned long long int num_sectors = geometry.kbToSector(get_size_k());
-	y2mil("num_sectors:" << num_sectors);
+	bool size_ok_for_msdos = num_sectors <= (1ULL << 32) - 1;
```
Comment (shundhammer) [inline storage/Devices/DiskImpl.cc]: Is there no nice predefined constant for this? No MAX_UINT32 or something like that?

File: storage/Devices/PartitionImpl.cc [hunk 4]
```diff
@@ -242,8 +243,26 @@ namespace storage
...
-	    // TODO look at id
-	    cmd_line += "ext2 ";
+	    switch (get_id())
+	    {
+		case ID_SWAP:
+		    cmd_line += "linux-swap ";
+		    break;
+
+		case ID_GPT_BOOT:
+		case ID_DOS16:
```
Comment (shundhammer) [inline storage/Devices/PartitionImpl.cc]: Always fat32, even for DOS16 partitions?

File: storage/Devices/DiskImpl.cc (hunk 2 above)
Comment (aschnell) [inline storage/Devices/DiskImpl.cc]: Yes, documented and unit-tested.

File: storage/Devices/PartitionImpl.cc (hunk 4 above)
Comment (aschnell) [inline storage/Devices/PartitionImpl.cc]: Copied from the old libstorage. AFAIS parted also supports fat16 and ntfs so I can add those.

**Comment by shundhammer** [comment]
LGTM now.

**Pull request merged**